zavod/zavod/validators/assertions.py

```diff
@@ -68,6 +68,9 @@ def feed(self, entity: Entity) -> None:
         self.stats.observe(entity)
 
     def finish(self) -> None:
+        if len(self.context.dataset.assertions) == 0:
+            self.context.log.warn("Dataset has no assertions.")
+
```

Comment:
could you break this out to its own PR since it's immediately useful and uncontroversial